ha/core/system_health/system_health.py

```diff
@@ -402,41 +427,72 @@ def process_event(self, healthevent: HealthEvent):
             component_id = healthevent.resource_id
             Log.info(f"SystemHealth: Processing {component}:{component_type}:{component_id} with status {status}")
 
+            # Update the node map
+            self.node_id = healthevent.node_id
+            self.node_map = {'cluster_id':healthevent.cluster_id, 'site_id':healthevent.site_id,
+                             'rack_id':healthevent.rack_id, 'storageset_id':healthevent.storageset_id}
+
             # Read the currently stored health value
             current_health = self.get_status_raw(component, component_id, comp_type=component_type,
                                         cluster_id=healthevent.cluster_id, site_id=healthevent.site_id,
                                         rack_id=healthevent.rack_id, storageset_id=healthevent.storageset_id,
                                         node_id=healthevent.node_id, server_id=healthevent.node_id,
                                         storage_id=healthevent.node_id)
 
+            current_timestamp = str(int(time.time()))
             if current_health:
+                current_health_dict = json.loads(current_health)
+                specific_info = current_health_dict["events"][0]["specific_info"]
+            if current_health and specific_info:
+                # If health is already stored and its a node_health, check further
+                stored_genration_id = current_health_dict["events"][0]["specific_info"]["generation_id"]
+                incoming_generation_id = healthevent.specific_info["generation_id"]
+                incoming_health_status = current_health_dict["events"][0]["status"]
+                pod_restart_val = current_health_dict["events"][0]["specific_info"]["pod_restart"]
                 # Update the current health value itself.
-                updated_health = EntityHealth.read(current_health)
+                latest_health = EntityHealth.read(current_health)
+                if stored_genration_id != incoming_generation_id:
+                    if incoming_health_status == status:
```

Comment:
Should there be a check of status == online additionally, to distinguish case of continuously failing pod probably sending failed status for same machine id but different generation id?

Reply:
Actual comment from Shailesh. Pasting it here for better understanding:
But if we have failed event in our current system health and receive failed again with diff generation id then it is some kind of crash loop with pod dying in later stages, when we do not need to send failed followed by online.

Reply:
Our main identifier for Node is machine id and our main check should rely on that I think. 
So, if:
MACHINE_ID: X, stored_status: failed, GEN_ID: Y
Now, MACHINE_ID X is trying to come up with say GEN_ID: A, B, C (And these are contineously failing).
For us That NODE is failed itself irrespective of what GENERATION_ID it is coming up..So, we don't need to handle this failure contineoulsy. Right?
Please let me know if there are still doubt. Thanks!

Reply:
That is inline with required logic, send failed->online only of online received with new generation-id for existing machine-id with online status. Please confirm if this is done.

Reply:
I think its not required. As dicussed in the DSU. We are inline with this.

Reply:
> That is inline with required logic, send failed->online only of online received with new generation-id for existing machine-id with online status. Please confirm if this is done.
Yes. Right

Reply:
Thanks for the confirmation and clarification.